core/restorer.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package core
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/kr/pretty"
+)
+
+var logger = loggo.GetLogger("juju-restore.core")
+
+// NewRestorer returns a new restorer for a specific database and
+// backup.
+func NewRestorer(db Database) *Restorer {
+	return &Restorer{db: db}
+}
+
+// Restorer checks the database health and backup file state and
+// restores the backup file into the database.
+type Restorer struct {
+	db Database
+}
+
+// CheckDatabaseState determines whether this database is appropriate
+// for restoring into.
+func (r *Restorer) CheckDatabaseState() error {
```

Comment:
Well, this is just checking that the database is in an appropriate state - checking whether a backup file should be restored into the database will be a separate step.

Reply:
Yess I agree that this is checking if db is in the right state. I am not saying that we should be checking backup file here and whether it can be restored into this mongo here. What I am  saying is that in level 2, when checking db state, we'll only be checking health of a non-replicaset mongo db since there will be only one member.

The logic you have written should still hold. But maybe worth a unit test with just one member?...

Reply:
Yup yup, I'll add one.